### DISCUSSION TITLE
How to update to new version?

### DISCUSSION THREAD

**Discussion by moebiussurfing** [open-webui/open-webui#7300] [Q&A] [answered]
Hello
I did the installation via `pip install open-webui`.
i don't see any script to update. I updated using pip, but I still getting the message offering the new version.
I use Windows 11.
thanks

**Comment by barryd15** [comment, marked as answer]
I had the same question and found my answer here:
Is there a tutroial how to update a pip install #7261
I had a pip install within a conda environment under WSL under Windows 11, and was able to adapt the instructions to my case. Your mileage may vary.

**Reply by DanBoyDan** [reply]
I have openwebui installed with miniconda and pip on clean Ubuntu 22.04 install, works fine. Yesterday received new version message, wanted to upgrade, found this page, followed all instructions,  used this command inside my conda env:

pip install --upgrade pip
pip install --upgrade open-webui

but does not work for me at all, it says:

(openwebui) dan@openwebui:~$ pip install open-webui --upgrade
Requirement already satisfied: open-webui in ./miniconda3/envs/openwebui/lib/python3.11/site-packages (0.4.7)
Requirement already satisfied: aiocache in ./miniconda3/envs/openwebui/lib/python3.11/site-packages (from open-webui) (0.12.3)

etc.

and it does not actually update anything, I am still on the older version.

**Reply by ay4t** [reply]
me, too i'm newbie on python world and don't know what to do

**Reply by Genine-Collin** [reply]
> I had the same question and found my answer here: Is there a tutroial how to update a pip install #7261 I had a pip install within a conda environment under WSL under Windows 11, and was able to adapt the instructions to my case. Your mileage may vary.

thx for the info

**Comment by silentoplayz** [comment]
https://docs.openwebui.com/

**Comment by sonjamichelle** [comment]
@silentoplayz  Your link does NOT solve the question that was asked. You merely regurgitated a link. did you even FULLY read the question?

@barryd15 Your answer and direction to #7261 DID answer the question and provide the CORRECT upgrade path for a pip install. Thank you!

**Comment by sonjamichelle** [comment]
@silentoplayz NO it does NOT.

`Installation with pip (Beta)
For users who prefer to use Python's package manager pip, Open WebUI offers a installation method. Python 3.11 is required for this method.

Install Open WebUI: Open your terminal and run the following command:

pip install open-webui

Start Open WebUI: Once installed, start the server using:

open-webui serve

This method installs all necessary dependencies and starts Open WebUI, allowing for a simple and efficient setup. After installation, you can access Open WebUI at http://localhost:8080/. Enjoy! 😄`

No where does it say to pass the `pip install --upgrade open-webui` command.

It ONLY explains the update process using docker and watchtower.

Feel free to post WHERE the command is found!

**Reply by silentoplayz** [reply]
@sonjamichelle A PR is welcome. I retracted my previous message because I realized I was wrong when looking at the documentation myself just a few minutes ago.

**Reply by sonjamichelle** [reply]
@silentoplayz  It's always wise to double-check what you are posting and ensure you avoid sharing a blind link without providing any context or explanation.

I searched the docs for a while before I resorted to a plain google search and came across this post.

**Reply by silentoplayz** [reply]
I don't need a lecture provided to me as a GitHub comment on what I should or shouldn't be doing. The person asked for something quite obvious to anyone who has used Python long enough (finding an answer as to how to upgrade a Python package is just a Google Search away too). And I didn't provide a URL blindly. I linked directly to the main page of Open WebUI's official documentation. All in all, I apologize that my regurgitated link didn’t prove to be helpful. I didn't write that section of the documentation and it’s all being written from the good will of contributors who wish to make things better and easier for us all in the long run.

**Reply by sif40** [reply]
Please be nice and respectful to each other and appreciate the countless of free time spent we all benefit from. And yes, it is frustrating to be urged to find of information, piece by piece, filtering the good stuff from the BS sprinkled everywhere

**Reply by richtong** [reply]
I’m
Happy to do a PR for this and also explain if you don’t want to create a vent or use system then

```
pipx install open-webi —python 3.11
pipx install open-webui —upgrade
```

**Reply by isaganijaen** [reply]
As of this date, the command ```pip install --upgrade open-webui``` works for me. I just had to close the open-web localhost:8080 and stop the terminal by hitting the ```ctrl+c``` where I ran the open-webui serve. 

Steps:
*(Do this inside the virtual environment if your initial installation was inside the virtual environment.)*
1. Make sure that open-webui is not running on the background. If it does, go to your terminal and hit ```ctrl+c```
2. run this command in your terminal ```pip install --upgrade open-webui```
3. Wait for the installation to complete
4. initiate the ```open-webui serve``` command again
5. Visit localhost:8080 and see if it worked for you.

**Reply by rgaricano** [reply]
Another good practice after updating is to clear your browser cache and site cookie, this prevents possible incorrect calls due to this previous caching.

**Comment by richtong** [comment]
The pipx is really important there.

Sent via Superhuman ( ***@***.*** )

**Comment by xflax** [comment]
After much back and forth, various commands that failed to update the version, and searching through the documentation, I found the simplest solution for me:

pip uninstall open-webui
pip install open-webui

It does what it's supposed to do, installs open-webui in the current version, and even retains the user data instead of deleting it. 
So if nothing else helps, this will get you where you want to go.

P.S.: Thanks for the tip about “--upgrade.” I kept getting a syntax error because of the missing hyphens.
I'll test it on the next update.
regards,
